Documentation/wip/support-tool.md

```diff
@@ -0,0 +1,62 @@
+# Fission Support Tool
+Fission now has rich functionality supported by multiple services, however, it brings the complexity of troubleshooting.
+This proposal tends to give a picture of fission support tool that can help both user and developer to locate the problem in short time.
+To achieve this, the support tool will dump related kubernetes objects, fission resources and pod logs from the given cluster.
+
+# Functionality
+
+## Environment Information Collection
+
+Before troubleshooting, some of the basic information is needed to give others an overview of kubernetes/fission user test with so that we can locate the problem in short time.
+
+* Fission version
+    * Client/Server version
+
+* Kubernetes cluster version
+    * Cluster version (i.e v1.9.7-gke.0)
+    * Running environment (i.e GKE, AKS and minikube)
+    * Nodes version and other information
+
+## Service Logs Collection
+
+The component logs and the logs of interaction between components are important for people to understand what really happened in cluster. Following are components need to collect logs from.
+
+* All fission component pods
+* Function pods
+* Builder pods
+* Environment pods
+
+## Object dumping
+
+Fission is deeply coupled with Kubernetes, most of the objects are created and maintained by it. There is two major type of objects need to be dumped from kubernetes:
+
+* K8S objects
+* CRD resources
+
+All objects should be dumped into a readable file format. It will be great if people can reproduce similar environment with these files.
+
+## Information upload
+
+Upload dump files to the specific backend server for support channel to analysis
+
+# CLI Interface
+
+```
+$ fission support collect
+  NAME:
+     fission support collect - Collect pod logs, fission resources and related kubernetes objects for troubleshooting
+  
+  USAGE:
+     fission support collect [command options] [arguments...]
```

Comment:
The sub-subcommand naming for this one feature is a bit awkward to me. 'collect' also does not immediately connects to the dumping of logs to me. Although---I can't really think of a better name right now. Maybe `fission support logs` or `fission support dump` or `fission support dump-logs`?

Reply:
I would then be in favor of using `fission support dump` as that best represents what happens with this command